AutoCollection/HttpDependencyParser.ts

```diff
@@ -114,6 +114,8 @@ class HttpDependencyParser extends RequestParser {
     private static _getUrlFromRequestOptions(options: any, request: http.ClientRequest) {
         if (typeof options === 'string') {
             options = url.parse(options);
+        } else if (options && typeof (url as any).URL === 'function' && options instanceof (url as any).URL) {
```

Comment:
I don't understand (url as any).URL, does `URL` exist on module `url`?

Reply:
@xiao-lix
It exists only in node 7+
https://nodejs.org/api/url.html#url_class_url

Since this SDK uses older node types, I `any` cast it for now.